cpp/lib/src/DbConnection.cc

```diff
@@ -449,6 +449,61 @@ void DbConnection::insertIntoTableOrDie(const std::string &table_name,
 }
 
 
+void DbConnection::insertIntoTableOrDie(const std::string &table_name, const std::vector<std::string> &column_names,
+                                        const std::vector<std::vector<std::string>> &values,
+                                        const DuplicateKeyBehaviour duplicate_key_behaviour, const std::string &where_clause)
+{
+    if (column_names.empty())
+        LOG_ERROR("at least one column name must be provided!");
+
+    if (not where_clause.empty() and duplicate_key_behaviour != DKB_REPLACE)
+        LOG_ERROR("\"where_clause\" is only valid when using the DKB_REPLACE mode!");
+
+    std::string insert_stmt(duplicate_key_behaviour == DKB_REPLACE ? "REPLACE" : "INSERT");
+    if (duplicate_key_behaviour == DKB_IGNORE)
+        insert_stmt += (type_ == T_MYSQL) ? " IGNORE" : " OR IGNORE";
```

Comment:
Should errors really generally be ignored or is it rather a "INSERT .. ON DUPLICATE KEY UPDATE" semantic intended?